### PR TITLE
fix/speech

### PR DESCRIPTION
**Pull request by NeonJarbas**
bumps ovos utils to include https://github.com/OpenVoiceOS/ovos_utils/pull/14

bumps ovos plugin manager to include https://github.com/OpenVoiceOS/OVOS-plugin-manager/pull/26

removes a hack with a TODO

refactors out the global config variable

wraps the fallback tts into a try: except:

### CODE REVIEW

File: mycroft/audio/speech.py [hunk 1]
```diff
@@ -44,12 +43,10 @@ def handle_speak(event):
...
     Parse sentences and invoke text to speech service.
     """
-    config = Configuration.get()
-    Configuration.set_config_update_handlers(bus)
-    global _last_stop_signal
+    global _last_stop_signal, tts
 
     # if the message is targeted and audio is not the target don't
-    # don't synthezise speech
+    # don't synthesise speech
```
Comment (ChanceNCounter) [inline mycroft/audio/speech.py:49]: In America it's 'synthesize'. Two nations divided by a common language 🤣

File: mycroft/audio/speech.py [hunk 2]
```diff
@@ -162,11 +127,15 @@ def mimic_fallback_tts(utterance, ident, listen):
...
-        LOG.debug("Mimic fallback, utterance : " + str(utterance))
-        tts.execute(utterance, ident, listen)
-    else:
-        LOG.error("TTS FAILURE! utterance : " + str(utterance))
+        try:
+            tts = _get_mimic_fallback()
+            LOG.debug("Mimic fallback, utterance : " + str(utterance))
+            tts.execute(utterance, ident, listen)
+            return
+        except:
```
Comment (ChanceNCounter) [inline mycroft/audio/speech.py:135]: Can we add a TODO here? Blanket catches mean somebody editing the called function is unable to crash loudly, but might not realize.

**Review by ChanceNCounter — COMMENTED** [review]
Goodbye, regex. I don't really get the removed hack. Was that compensating for Picroft, Pi, or Raspbian, and do we care?

File: mycroft/audio/speech.py (hunk 2 above)
Comment (NeonJarbas) [reply mycroft/audio/speech.py:135]: will soon be deprecated in a follow up PR to make fallback configurable. it prints a log below about the error also

File: mycroft/audio/speech.py (hunk 1 above)
Comment (NeonJarbas) [reply mycroft/audio/speech.py:49]: pycharm autocorrected it!

**Review by j1nx — COMMENTED** [review]
OK, will follow all related PR's, bump, test and report back.

**Comment by j1nx** [comment]
Testing this PR however following error is for all TTS output.

```
2022-02-09 19:56:15.139 - OVOS - mycroft.audio.speech:mute_and_speak:104 - ERROR - TTS execution failed.
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/mycroft/audio/speech.py", line 99, in mute_and_speak
    tts.execute(utterance, ident, listen)
  File "/usr/lib/python3.8/site-packages/ovos_plugin_manager/templates/tts.py", line 418, in execute
    self._execute(sentence, ident, listen, **kwargs)
  File "/usr/lib/python3.8/site-packages/ovos_plugin_manager/templates/tts.py", line 449, in _execute
    audio_file, phonemes = self._synth(sentence, sentence_hash, **kwargs)
  File "/usr/lib/python3.8/site-packages/ovos_plugin_manager/templates/tts.py", line 506, in _synth
    audio.path, phonemes = self.get_tts(sentence, str(audio), **kwargs)
  File "/usr/lib/python3.8/site-packages/ovos_tts_plugin_mimic2/__init__.py", line 43, in get_tts
    results = requests.get(self.url, params=params).json()
  File "/usr/lib/python3.8/site-packages/requests/models.py", line 900, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2022-02-09 19:56:15.216 - OVOS - ovos_plugin_manager.templates.tts:_play:118 - ERROR - 'NoneType' object has no attribute 'split'
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/ovos_plugin_manager/templates/tts.py", line 107, in _play
    self.p = play_audio(data)
  File "/usr/lib/python3.8/site-packages/ovos_utils/sound/__init__.py", line 22, in play_audio
    ext = uri.split("?")[0].split(".")[-1]
AttributeError: 'NoneType' object has no attribute 'split'
```
Used the default mycroft setup route utilizing the Mimic2 and selene pairing.

**Comment by j1nx** [comment]
Following error is within voice.log but not sure if this related.
```
2022-02-09 19:58:47.774 - OVOS - mycroft.client.speech.listener:run:396 - DEBUG - Config has changed, reloading...
2022-02-09 19:58:47.817 - OVOS - mycroft.client.speech.listener:run:97 - ERROR - Exception in AudioProducer
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/mycroft/client/speech/listener.py", line 72, in run
    audio, lang = self.loop.responsive_recognizer.listen(
  File "/usr/lib/python3.8/site-packages/mycroft/client/speech/mic.py", line 802, in listen
    ww_data, lang = self._wait_until_wake_word(source, sec_per_buffer)
TypeError: cannot unpack non-iterable NoneType object
```

File: mycroft/audio/speech.py [hunk 3]
```diff
@@ -62,51 +59,18 @@ def handle_speak(event):
...
         ident = 'unknown'
 
-    start = time.time()  # Time of speech request
     with lock:
         stopwatch = Stopwatch()
         stopwatch.start()
+
         utterance = event.data['utterance']
         listen = event.data.get('expect_response', False)
-        # This is a bit of a hack for Picroft.  The analog audio on a Pi blocks
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply mycroft/audio/speech.py:71]: not an issue, i believe this was due to lack of pulse audio in very old raspbian

plugins like mimic2 already do this internally and better, i see no reason to keep this around at all

**Comment by JarbasAl** [comment]
@j1nx 

thats mimic2 plugin failing, all the errors are just logs but actually handled in code

the second thing is a new issue, #50

**Pull request merged**